### ISSUE TITLE
Code block contrast ratio

### ISSUE THREAD

**Issue by willhack** [vuejs/vitepress#4486]
### Describe the bug

While auditing another site (which uses vitepress), I noticed the contrast ratio on code languages and line numbers doesn't meet WCAG guidelines.

<img width="700" alt="" src="https://github.com/user-attachments/assets/84ef9a18-94b2-41ad-bf0a-42991f01a5cb" />

<img width="700" alt="" src="https://github.com/user-attachments/assets/a0d88f61-9edd-4d28-b60e-a6b8f5d25321" />

Looking back to color system PR [#2797](https://github.com/vuejs/vitepress/pull/2797) it seems as though `--vp-c-text-3` was not intended to be used with `--vp-c-bg-alt`

> XXX-3: The color for solid background, such as bg color of the button. It must satisfy the contrast ratio with pure white (#ffffff) text on top of it

I have a possible fix available and will submit a PR

### System Info

```Text
System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Max
    Memory: 5.63 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Chrome: 131.0.6778.265
    Edge: 131.0.2903.112
    Safari: 17.5
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
